configuration/configuration.py

```diff
@@ -199,6 +199,17 @@ def read_secret(secret_name):
 # this setting is derived from the installed location.
 SCRIPTS_ROOT = os.environ.get('SCRIPTS_ROOT', '/etc/netbox/scripts')
 
+# The backend storage engine for handling uploaded files (e.g. image attachments). NetBox supports integration with
+# the django-storages package, which provides backends for several popular file storage services. If not configured,
+# local filesystem storage will be used.
+# The configuration parameters for the specified storage backend are defined under the STORAGE_CONFIG setting.
+STORAGE_BACKEND = os.environ.get('STORAGE_BACKEND', None)
+
+# A dictionary of configuration parameters for the storage backend configured as STORAGE_BACKEND. The specific
+# parameters to be used here are specific to each backend; see the django-storages documentation for more detail.
+# If STORAGE_BACKEND is not defined, this setting will be ignored.
+STORAGE_CONFIG = os.environ.get('STORAGE_CONFIG', None)
```

Comment:
> A dictionary of configuration parameters for the storage backend configured as STORAGE_BACKEND

This says the value of the STORAGE_CONFIG should be a dictionary. But ` os.environ.get` will return only strings. Do you have an example how to use it?

Reply:
Searching for "STORAGE_CONFIG" [here](https://netbox.readthedocs.io/en/stable/release-notes/version-2.7/), shows an example, the expected syntax is like this one:
```
    STORAGE_CONFIG = {
    'AWS_ACCESS_KEY_ID': '<Key>',
    'AWS_SECRET_ACCESS_KEY': '<Secret>',
    'AWS_STORAGE_BUCKET_NAME': 'netbox',
    'AWS_S3_REGION_NAME': 'eu-west-1',
}
```

I added this config using extraEnvs for the helm values:
```yaml
extraEnvs:
  LOGIN_REQUIRED: true
  METRICS_ENABLED: true
  STORAGE_BACKEND: 'storages.backends.s3boto3.S3Boto3Storage'
  STORAGE_CONFIG: |
    {
    "AWS_S3_ENDPOINT_URL": "http://netbox-minio:9000",
    "AWS_S3_USE_SSL": False,
    "AWS_ACCESS_KEY_ID": "minio-access-key",
    "AWS_SECRET_ACCESS_KEY": "minio-secret-key",
    "AWS_STORAGE_BUCKET_NAME": "netbox-assets"
    }
```

The env vars values in the pod seemed correct, but I could not properly test them.

Reply:
I don't mean to be rude, but did you successfully test the proposed feature before submitting it? Because in my understanding I don't see how it could work as it is.

In Python (`configuration.py` is just Python code), `os.environ.get` will return a [`string`](https://docs.python.org/3/library/string.html). But `STORAGE_CONFIG` is expected to be a [`Dictionary`](https://docs.python.org/3/tutorial/datastructures.html#dictionaries). Plus I suspect that `STORAGE_BACKEND` is expected to be a `class`, and not _the name of a class as `string`_.

Make no mistake, I could be wrong and just don't get it. That's why I ask you to either provide a working example (using the minio container seems a good idea towards that) or to amend your PR so that we get to a working configuration.

Reply:
Don't worry, you were not rude at all and in fact you were right:  the STORAGE_CONFIG needs to be converted to a Dictionary.
The STORAGE_BACKEND appears to be correct though.
As I said before, I could test the vars values inside the k8s pods but I could not create an updated build incorporating the changes.

I have made some changes to the patch and I'll try to check everything on a new container image build and testing it with "plain" docker, asap.
Thanks for the feedback.

Reply:
Hello again, I properly tested the feature and I confirm that it works as expected.
There's a minor issue: I had to add "boto3" to the requirements in the Dockerfile, but any backend supported by django-storages would require its own libraries.

Let me know if i could help any further.